plugin/pkg/admission/limitranger/admission.go

```diff
@@ -561,48 +525,39 @@ func PodValidateLimitFunc(limitRange *corev1.LimitRange, pod *api.Pod) error {
 
 		// enforce pod limits on init containers
 		if limitType == corev1.LimitTypePod {
-			containerRequests, containerLimits := []api.ResourceList{}, []api.ResourceList{}
-			for j := range pod.Spec.Containers {
-				container := &pod.Spec.Containers[j]
-				containerRequests = append(containerRequests, container.Resources.Requests)
-				containerLimits = append(containerLimits, container.Resources.Limits)
+			v1Pod := &corev1.Pod{}
+			// these conversion functions should never error
+			if err := k8s_api_v1.Convert_core_Pod_To_v1_Pod(pod, v1Pod, nil); err != nil {
+				errs = append(errs, err)
+				return utilerrors.NewAggregate(errs)
 			}
-			podRequests := sum(containerRequests)
-			podLimits := sum(containerLimits)
-			for j := range pod.Spec.InitContainers {
-				container := &pod.Spec.InitContainers[j]
-				// take max(sum_containers, any_init_container)
-				for k, v := range container.Resources.Requests {
-					if v2, ok := podRequests[k]; ok {
-						if v.Cmp(v2) > 0 {
-							podRequests[k] = v
-						}
-					} else {
-						podRequests[k] = v
-					}
-				}
-				for k, v := range container.Resources.Limits {
-					if v2, ok := podLimits[k]; ok {
-						if v.Cmp(v2) > 0 {
-							podLimits[k] = v
-						}
-					} else {
-						podLimits[k] = v
-					}
-				}
+
+			// pod hasn't scheduled so we don't care about InPlacePodVerticalScalingEnabled
+			opts := resourcehelper.PodResourcesOptions{}
+			podRequests := resourcehelper.PodRequests(v1Pod, opts)
+			podLimits := resourcehelper.PodLimits(v1Pod, opts)
+			var apiPodRequests, apiPodLimits api.ResourceList
+			if err := k8s_api_v1.Convert_v1_ResourceList_To_core_ResourceList(&podRequests, &apiPodRequests, nil); err != nil {
+				errs = append(errs, err)
+				return utilerrors.NewAggregate(errs)
+			}
+			if err := k8s_api_v1.Convert_v1_ResourceList_To_core_ResourceList(&podLimits, &apiPodLimits, nil); err != nil {
```

Comment:
double-converting this back and forth on a high velocity API like pods is really unfortunate. do we know the performance impact of these? unless this is fixing a bug, I'd be inclined to revert the changes to this file and add a TODO to follow up to try to find a way to collapse to the helpers without paying the conversion cost

Reply:
I didn't see a good alternative other than writing the calculation again due to the different types.  I don't think there's a bug in the code at the moment since resizing can't occur prior to scheduling, but this calculation will need to be updated with the sidecar KEP code.